[MTM-54377] Smart rules documentation update (#2458)
SoftwareAG/c8y-docs: content/cockpit/smart-rules.md

- **Object parameters** are stored in the group or device. These parameters can be edited after the smart rule has been created. An example includes min and max values for thresholds.

There are two different types of smart rules:
- Global smart rules
- Local smart rules

**Global smart rules**

>   Global smart rules are created in a global context (**Smart rules** page, alarms, data explorer, and so on). The rule that created as global monitors whole inventory (any asset type, eg: groups, devices, configurations, applications, etc).

Reviewer: ```suggestion
  Global smart rules are created in a global context (**Smart rules** page, alarms, data explorer, and so on). Rules created as global monitor the whole inventory (any asset type such as groups, devices, configurations, applications).
```

- **Object parameters** are stored in the group or device. These parameters can be edited after the smart rule has been created. An example includes min and max values for thresholds.

There are two different types of smart rules:
- Global smart rules
- Local smart rules

**Global smart rules**

>   Global smart rules are created in a global context (**Smart rules** page, alarms, data explorer, and so on).

Author: Just thought when checking the task description that this part may be expanded like

```suggestion
  Global smart rules are created in a global context (**Smart rules** page, alarms, data explorer, and so on). The rule that created as global monitors whole inventory (any asset type, eg: groups, devices, configurations, applications, etc).
```

- To delete a global smart rule: ADMIN permission for permission type "Global smart rules" or "CEP management" and ADMIN permission for permission type "Inventory"
{{< /c8y-admon-req >}}

> Depending on initial configuration provided when creating a global smart rule there are 2 types possible.

Reviewer 2: ```suggestion
Depending on initial configuration provided when creating a global smart rule there are two types possible:
```

> Depending on initial configuration provided when creating a global smart rule there are two types possible:

Reviewer: ```suggestion
Depending on the initial configuration provided when creating a global smart rule there are two types possible:
```

> Depending on the initial configuration provided when creating a global smart rule there are two types possible:

Reviewer 3: ```suggestion
Depending on the initial configuration of a global smart rule, there are two types possible:
```

> - Global smart rules by default enabled for all assets: When no asset is selected with the configuration. Such smart rule is automatically applied to any asset available in the system including those assets that are added later in time.

Reviewer 3: ```suggestion
- Global smart rules enabled for all assets by default: When no asset is selected in the initial configuration. Such a smart rule is automatically applied to any asset available in the system, including assets added later in time.
```

> - global smart rules by default enabled for all assets - when by providing the configuration no asset is selected. Such smart rule is automatically applied to any asset available in the system including those assets that a added later in time.

Reviewer 4: ```suggestion
- Global smart rules by default enabled for all assets: When by providing the configuration no asset is selected. Such smart rule is automatically applied to any asset available in the system including those assets that are added later in time.
```

> - Global smart rules by default enabled for all assets: When by providing the configuration no asset is selected. Such smart rule is automatically applied to any asset available in the system including those assets that are added later in time.

Reviewer: ```suggestion
- Global smart rules by default enabled for all assets: When no asset is selected with the configuration. Such smart rule is automatically applied to any asset available in the system including those assets that are added later in time.
```

> - global smart rules by default disabled for all assets - when by providing configuration at least one asset is selected. Such smart rule is automatically disabled to aby other asset except those provided initially in configuration. It can be enabled for newly added assets manually.

Reviewer 2: ```suggestion
- Global smart rules by default disabled for all assets: When by providing configuration at least one asset is selected. Such smart rule is automatically disabled to any other asset except those provided initially with the configuration. You can manually enable it for newly added assets.
```

> - Global smart rules by default disabled for all assets: When at least one asset is selected with the configuration. Such smart rule is automatically disabled to any other asset except those provided initially with the configuration. You can manually enable it for newly added assets.

Reviewer 3: ```suggestion
- Global smart rules disabled for all assets by default: When at least one asset is selected in the initial configuration. Such a smart rule is automatically disabled for any other assets except the ones selected during creation. You can manually enable it for other existing or newly added assets.
```

> - Global smart rules by default disabled for all assets: When by providing configuration at least one asset is selected. Such smart rule is automatically disabled to any other asset except those provided initially with the configuration. You can manually enable it for newly added assets.

Reviewer: ```suggestion
- Global smart rules by default disabled for all assets: When at least one asset is selected with the configuration. Such smart rule is automatically disabled to any other asset except those provided initially with the configuration. You can manually enable it for newly added assets.
```

**Local smart rules**

>   Local smart rules are created in either a group or a device. They are visible to everyone with access to the group/device.

Author: ```suggestion
  Local smart rules are created in either a group or a device. They are visible to everyone with access to the group/device.
  Those rules affecting current asset in context of which are created and may affect any child asset depending on configuration provided. 
```

Reviewer: A local rule only affects the asset in which it was created and may affect all child assets depending on the configuration.

- To delete a global smart rule: ADMIN permission for permission type "Global smart rules" or "CEP management" and ADMIN permission for permission type "Inventory"
{{< /c8y-admon-req >}}

Depending on the initial configuration of a global smart rule, there are two types possible:

> - Global smart rules enabled for all assets by default: When no asset is selected in the initial configuration. Such a smart rule is automatically applied to any asset available in the system, including assets added later in time.

Reviewer 5: to any asset => to each asset?

Reviewer 2: From following perspective I think that current form is correct: 

`knowing that basket have 5 red and 5 blue assets, pick any asset and check whenever it is red` 

sounds more correct then 

`knowing that basket have 5 red and 5 blue assets, pick each asset and check whenever it is red`

Reviewer: I agree, it must be "each asset" if you want it to be applied to all assets and not only to one.

Reviewer 2: outvoted XD so changed.

- Global smart rules disabled for all assets by default: When at least one asset is selected in the initial configuration. Such a smart rule is automatically disabled for any other assets except the ones selected during creation. You can manually enable it for other existing or newly added assets.

**Local smart rules**

>   Local smart rules are created in either a group or a device. A local rule only affects the asset in which it was created and may affect all child assets depending on the configuration.

Reviewer 5: in either a group or a device => for a  group or a device

Reviewer 2: It may be created also for something else (in theory), that might not be group or device. So `Created in` seems more correct for this case since we want to point in what context we currently are, but not for what context rule is configured.

Reviewer: I would actually say "from a group or device" :-)

Reviewer 2: ok :-)

- To delete a global smart rule: ADMIN permission for permission type "Global smart rules" or "CEP management" and ADMIN permission for permission type "Inventory"
{{< /c8y-admon-req >}}

Depending on the initial configuration of a global smart rule, there are two types possible:
- Global smart rules enabled for all assets by default: When no asset is selected in the initial configuration. Such a smart rule is automatically applied to any asset available in the system, including assets added later in time.
- Global smart rules disabled for all assets by default: When at least one asset is selected in the initial configuration. Such a smart rule is automatically disabled for any other assets except the ones selected during creation. You can manually enable it for other existing or newly added assets.

**Local smart rules**

  Local smart rules are created in either a group or a device. A local rule only affects the asset in which it was created and may affect all child assets depending on the configuration.

>   Those rules affecting current asset in context of which are created and may affect any child asset depending on configuration provided. 

Reviewer 3: This sentence sounds duplicate?

Reviewer 2: yeah, probably mistake during patching

Reviewer: @iiar-c8y  Please remove the second sentence.

Reviewer 2: done

- To delete a local smart rule: ADMIN permission for permission type "Inventory" or CHANGE permission for "Inventory" in the inventory roles
{{< /c8y-admon-req >}}

> Smart rules can be seen in three places:

Reviewer: Only 2 list items have a bullet, which one is the third one? Should get a list format as well then (-)

Reviewer 2: right, this is the third one: `in the **Status** tab of the owned microservice`.

Reviewer 2: fixed, please check

  In a local context (group or device) the local smart rules are shown. For users with the relevant permissions, both local and global smart rules are shown.

>   From the context of a group or device, or in the **Status** tab of the owned microservice you can only modify  local smart rules. Editing global smart rules is only possible on the **Global smart rules** page. 

Reviewer: ```suggestion
  From the context of a group or device, or in the **Status** tab of the owned microservice, you can only modify local smart rules. Editing global smart rules is only possible on the **Global smart rules** page. 
```

>   From the context of a group or device, or in the **Status** tab of the owned microservice it is possible to modify only local smart rule. Editing global smart rules is only possible on the **Global smart rules** page. 

Reviewer: ```suggestion
  From the context of a group or device, or in the **Status** tab of the owned microservice you can only modify  local smart rules. Editing global smart rules is only possible on the **Global smart rules** page. 
```

>   The microservice details show the global and local smart rules (considering the permissions of the user). From this page it is possible to configure a limited number of smart rules, mainly related to alarms that might be created by the microservice while transitioning via different deployment states.

Reviewer 3: Might not be clear that we still talk about "Status" tab:
```suggestion
  The "Status" tab shows both the global and local smart rules (considering the permissions of the user). In this tab you can only configure smart rules that work with alarms that are created by the microservice while transitioning through different deployment states.
```

Reviewer 2: Partially applied, used `Here` instead of `In this tab` to not repeat myself. Besides didn't use `you` in sentence.

  In a local context (group or device) the local smart rules are shown. For users with the relevant permissions, both local and global smart rules are shown.

>   From the context of a group or device it is possible to modify only local smart rule. Editing global smart rules is only possible on the **Global smart rules** page in the **Status** tab of the owned microservice. 

Reviewer: ```suggestion
  From the context of a group or device it is only possible to modify local smart rules. Editing global smart rules is only possible in the **Global smart rules** page in the **Status** tab of the owned microservice. 
```

Reviewer 2: this was wrong, changed content.

>   The microservice page shows the global and local smart rules (considering the permissions of the user). From this page it is possible to configure a limited number of smart rules, mainly related to alarms that might be created by the microservice while transitioning via different deployment states.

Reviewer: ```suggestion
  The microservice details show the global and local smart rules (considering the permissions of the user). From this page it is possible to configure a limited number of smart rules, mainly related to alarms that might be created by the microservice while transitioning via different deployment states.
```

>   From the microservice page it is also possible to modify only local smart rules. To edit global smart rules navigate to the **Global smart rules** page.

Reviewer: ```suggestion
  From the microservice details it is also possible to modify only local smart rules. To edit global smart rules navigate to the **Global smart rules** page.
```

Author: This sentence was removed, since similar content already mentioned above.

  In a local context (group or device) the local smart rules are shown. For users with the relevant permissions, both local and global smart rules are shown.

>   From the context of a group or device it's possible to modify only local smart rules. Editing global smart rules allowed only from global smart rules view.

Reviewer 4: ```suggestion
  From the context of a group or device it is possible to modify only local smart rule. Editing global smart rules is only possible on the **Global smart rules** page in the **Status** tab of the owned microservice. 
```

> - In the **Status** tab of owned microservice.

Reviewer 4: ```suggestion
```

>   In a local context of microservice are shown global and local smart rules (considering the permissions of the user). From the view it is possible to configure limited number of smart rules, mainly related to alarms that might be created by microservice while transitioning via different deployment states.

Reviewer 4: ```suggestion
  The microservice page shows the global and local smart rules (considering the permissions of the user). From this page it is possible to configure a limited number of smart rules, mainly related to alarms that might be created by the microservice while transitioning via different deployment states.
```

>   From context of microservice it is possible to modify only local smart rules. Editing global smart rules allowed only from global smart rules view.

Reviewer 4: ```suggestion
  From the microservice page it is also possible to modify only local smart rules. To edit global smart rules navigate to the **Global smart rules** page.
```

  In a local context (group or device) the local smart rules are shown. For users with the relevant permissions, both local and global smart rules are shown.

- In the **Status** tab of the owned microservice.

>   The microservice details show the global and local smart rules (considering the permissions of the user). Here it is possible to configure a limited number of smart rules, mainly related to alarms that might be created by the microservice while transitioning through different deployment states.

Reviewer 3: "a limited number of smart rules" suggests a constraint on the number of created smart rules, while you rather mean a constraint on what types of smart rules can be created in this place.

Reviewer 2: improved
